src/pytorch_lightning/overrides/data_parallel.py

```diff
@@ -65,7 +65,7 @@ def forward(self, *inputs: Any, **kwargs: Any) -> Any:
         output = super().forward(*inputs, **kwargs)
 
         def output_transform(data: Any) -> Any:
-            device = cast(torch.device, self.module.device)
+            device = self.module.device
```

Comment:
resolved redundant cast after setting DeviceDtypeModuleMixin.device to torch.device